### ISSUE TITLE
Error while running Rails/Blank on master.

### ISSUE THREAD

**Issue by Mihail-K** [bbatsov/rubocop#4814]
## Expected behaviour

No error should occur.
No offence should be reported on the line as it does not use the `.present?` or `.blank?` methods, or anything that should be converted to them.

## Actual behaviour

Crash occurs in cop `Rails/Blank`:
```
An error occurred while Rails/Blank cop was inspecting /Users/mintyfresh/-omitted-/app/models/ban.rb:51:42.
undefined method `type' for nil:NilClass
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/cop/rails/blank.rb:50:in `nil_or_empty?'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/cop/rails/blank.rb:86:in `on_or'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/cop/commissioner.rb:44:in `block (2 levels) in on_or'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/cop/commissioner.rb:109:in `with_cop_error_handling'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/cop/commissioner.rb:43:in `block in on_or'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/cop/commissioner.rb:42:in `each'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/cop/commissioner.rb:42:in `on_or'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/ast/traversal.rb:154:in `on_block'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/cop/commissioner.rb:48:in `on_block'
(eval):2:in `block in on_pair'
(eval):2:in `each'
(eval):2:in `on_pair'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/cop/commissioner.rb:48:in `on_pair'
(eval):2:in `block in on_hash'
(eval):2:in `each'
(eval):2:in `on_hash'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/cop/commissioner.rb:48:in `on_hash'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/ast/traversal.rb:101:in `block in on_send'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/ast/traversal.rb:99:in `each'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/ast/traversal.rb:99:in `each_with_index'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/ast/traversal.rb:99:in `on_send'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/cop/commissioner.rb:48:in `on_send'
(eval):2:in `block in on_begin'
(eval):2:in `each'
(eval):2:in `on_begin'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/cop/commissioner.rb:48:in `on_begin'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/ast/traversal.rb:88:in `on_class'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/cop/commissioner.rb:48:in `on_class'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/ast/traversal.rb:12:in `walk'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/cop/commissioner.rb:60:in `investigate'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/cop/team.rb:114:in `investigate'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/cop/team.rb:102:in `offenses'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/cop/team.rb:44:in `inspect_file'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/runner.rb:258:in `inspect_file'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/runner.rb:205:in `block in do_inspection_loop'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/runner.rb:237:in `block in iterate_until_no_changes'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/runner.rb:230:in `loop'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/runner.rb:230:in `iterate_until_no_changes'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/runner.rb:201:in `do_inspection_loop'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/runner.rb:111:in `block in file_offenses'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/runner.rb:121:in `file_offense_cache'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/runner.rb:109:in `file_offenses'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/runner.rb:100:in `process_file'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/runner.rb:78:in `block in each_inspected_file'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/runner.rb:75:in `each'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/runner.rb:75:in `reduce'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/runner.rb:75:in `each_inspected_file'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/runner.rb:67:in `inspect_files'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/runner.rb:39:in `run'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/cli.rb:86:in `execute_runner'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/lib/rubocop/cli.rb:32:in `run'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/bin/rubocop:13:in `block in <top (required)>'
/Users/mintyfresh/.rvm/rubies/ruby-2.3.4/lib/ruby/2.3.0/benchmark.rb:308:in `realtime'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bundler/gems/rubocop-d5e221dad130/bin/rubocop:12:in `<top (required)>'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bin/rubocop:23:in `load'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bin/rubocop:23:in `<main>'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bin/ruby_executable_hooks:15:in `eval'
/Users/mintyfresh/.rvm/gems/ruby-2.3.4/bin/ruby_executable_hooks:15:in `<main>'
.

1 file inspected, no offenses detected
```

## Steps to reproduce the problem

`app/models/ban.rb:51:42` which is the offending line, assuming the stack trace is correct:
```ruby
before_save :remove_user_role, if: -> { expired_changed?(to: true) || deleted_changed?(to: true) }
```

## RuboCop version

```
$ rubocop -V
0.50.0 (using Parser 2.4.0.0, running on ruby 2.3.4 x86_64-darwin16)
```

Specifically pointing to `master @ d5e221dad1308cc5ac40735170db6816bdd631a0`

**Comment by pocke** [comment]
Minimum code to reproduce:

```ruby
x(1) || something
```


```bash
$ rubocop -R -d  --cache false
```
